pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1978,6 +1980,54 @@ static __always_inline int should_submit(u32 event_id, config_entry_t *config)
 
 // INTERNAL: BUFFERS -------------------------------------------------------------------------------
 
+static __always_inline int cache_event_data(event_data_t *data)
+{
+    u32 zero = 0;
+    event_data_t *presv_data = bpf_map_lookup_elem(&cached_event_data_map, &zero);
+    if (presv_data == NULL)
+        return -1;
+
+    return bpf_probe_read(presv_data, sizeof(event_data_t), data);
+}
+
+static __always_inline int get_cached_event_data(event_data_t *data, void *ctx)
+{
+    if (data == NULL) {
+        return -1;
+    }
+    u32 zero = 0;
+    event_data_t *presv_data = bpf_map_lookup_elem(&cached_event_data_map, &zero);
+    if (presv_data == NULL)
+        return -2;
+    int read_ret = bpf_probe_read(data, sizeof(event_data_t), presv_data);
+    if (read_ret)
+        return -3;
+
+    data->ctx = ctx;
+
+    data->config = bpf_map_lookup_elem(&config_map, &zero);
+    if (unlikely(data->config == NULL))
+        return -4;
+
+    int buf_idx = SUBMIT_BUF_IDX;
+    data->submit_p = bpf_map_lookup_elem(&bufs, &buf_idx);
+    if (unlikely(data->submit_p == NULL))
+        return -5;
+
+    bool task_info_initalized = false;
+    data->task_info = init_task_info(data->context.task.host_tid, &task_info_initalized);
```

Comment:
Is there a reason to do this again in the tail call? Won't it be the same as what we've loaded?